src/deploy/functions/runtimes/node/index.ts

```diff
@@ -66,13 +66,75 @@ export class Delegate {
   // to decide whether to use the JS export method of discovery or the HTTP container contract
   // method of discovery.
   _sdkVersion: string | undefined = undefined;
-  get sdkVersion() {
+  get sdkVersion(): string {
     if (this._sdkVersion === undefined) {
       this._sdkVersion = versioning.getFunctionsSDKVersion(this.sourceDir) || "";
     }
     return this._sdkVersion;
   }
 
+  _bin = "";
```

Comment:
Can you write some docs on this (+ other functions) ?

Reply:
Most of these fields are documented in the abstract class. I'd prefer if we don't duplicate the comments?

https://github.com/firebase/firebase-tools/blob/e7d25e292746132fd6084b027fec010c1d912a1a/src/deploy/functions/runtimes/index.ts#L65-L68